topomath eq editor updates #339, #345, #360

riteshsamala opened the pull request.

jonwetzel commented on www/js/con-author.js:
```diff
@@ -795,6 +890,22 @@ define([
+					schemaTabOb[parSchema].forEach(function(atomSchema){
+								var sname = atomSchema["Name"];
+								var obj = {value: sname, label: sname};
+								schemaWidget.addOption(obj);
```
There's a bug with this: we add the options every time this function is called, but we never delete them.  So an extra copy of the schemas is being added every time the node editor opens.  

Perhaps adding the options should be done once when the tutor first loads (in index.php or main.js?) instead of when the node editor opens, because we don't expect the list of schemas to change during a session.

jonwetzel commented on www/js/controller.js:
```diff
@@ -412,9 +415,9 @@ define([
 			}));
 
 			 //event handler for quantity node description field
```
Update comment to match code.

jonwetzel commented on www/js/equation.js:
```diff
@@ -43,6 +43,8 @@ define([
 			try{
 				array.forEach(eqs, function(eq, count){
 					expressions[count] = Parser.parse(eq);
+					console.log("expr array",expressions);
```
Remove this console.log if no longer useful.

jonwetzel commented on www/schema-table.json:
```diff
@@ -7,7 +7,8 @@
 			},
-			"Comments": "The whole is equal to the sum of two parts."
+			"Comments": "The whole is equal to the sum of two parts.",
+			"slot names": "Whole, Part A, Part B"
```
As previously discussed, the slot names field is redundant with the Mapping.  Go ahead and remove all "slot names" fields.

riteshsamala commented:
I have updated the code in latest commit on current branch. It also includes enhancements. I also found one bug while testing, entity text box was accepting empty entity input value before semicolon like ";entity1". Fixed it as well.
